logstash-core/spec/api/lib/api/plugins_spec.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+require_relative "../../spec_helper"
+require "sinatra"
+require "app/modules/plugins"
+require "logstash/json"
+
+describe LogStash::Api::Plugins do
+
+  include Rack::Test::Methods
+
+  def app()
+    described_class
+  end
+
+  before(:all) do
+    get "/"
+  end
+
+  let(:payload) { JSON.parse(last_response.body) }
+
+  it "respond to plugins resource" do
+    expect(last_response).to be_ok
+  end
+
+  it "return valid json content type" do
+    expect(last_response.content_type).to eq("application/json")
+  end
+
+  it "return the expected schema" do
+    expect(payload.keys).to include("plugins", "total")
+    payload["plugins"].each do |plugin|
+      expect(plugin.keys).to include("name", "version")
```

Comment:
yes it makes the test pas, but this situation is impossible to happen for how, if spec.version.to_s return nil, this means we installed plugins without version ... this will always return.